### PR TITLE
認証管理をsorceryに変更

### PR DESCRIPTION
**Pull request by segawa-sh**

### CODE REVIEW

File: app/views/kaminari/_paginator.html.erb [hunk 1]
```diff
@@ -0,0 +1,15 @@
+<%= paginator.render do -%>
+  <ul class="pagination">
+    <%= first_page_tag unless current_page.first? %>
+    <%= prev_page_tag unless current_page.first? %>
+    <% each_page do |page| -%>
```
Comment (tektoh) [inline app/views/kaminari/_paginator.html.erb]: __[rails_best_practices]__
* move code into model (page use_count > 2) [:ref](http://rails-bestpractices.com/posts/2010/07/24/move-code-into-model/)

File: config/routes.rb [hunk 2]
```diff
@@ -1,11 +1,17 @@
 Rails.application.routes.draw do
-  devise_for :users
-
   if Rails.env.development?
     get '/login_as/:user_id', to: 'development/sessions#login_as'
     mount LetterOpenerWeb::Engine, at: '/letter_opener'
   end
 
+  resource :users do
```
Comment (segawa-sh) [inline config/routes.rb:7]: **[rails_best_practices]**
- restrict auto-generated routes users (only: []) [:ref](http://rails-bestpractices.com/posts/2011/08/19/restrict-auto-generated-routes/)

File: config/initializers/sorcery.rb [hunk 3]
```diff
@@ -0,0 +1,464 @@
...
+# Here you can configure each submodule's features.
+Rails.application.config.sorcery.configure do |config|
+  # -- core --
+  # What controller action to call for non-authenticated users. You can also
+  # override the 'not_authenticated' method of course.
+  # Default: `:not_authenticated`
+  #
+  # config.not_authenticated_action =
+
+
```
Comment (segawa-sh) [inline config/initializers/sorcery.rb]: **[rubocop]**
- Style/EmptyLines: Extra blank line detected.

File: config/initializers/sorcery.rb [hunk 4]
```diff
@@ -0,0 +1,464 @@
...
+  # When a non logged in user tries to enter a page that requires login, save
+  # the URL he wanted to reach, and send him there after login, using 'redirect_back_or_to'.
+  # Default: `true`
+  #
+  # config.save_return_to_url =
+
+
```
Comment (segawa-sh) [inline config/initializers/sorcery.rb]: **[rubocop]**
- Style/EmptyLines: Extra blank line detected.

File: config/initializers/sorcery.rb [hunk 5]
```diff
@@ -0,0 +1,464 @@
...
+  # Set domain option for cookies; Useful for remember_me submodule.
+  # Default: `nil`
+  #
+  # config.cookie_domain =
+
+
```
Comment (segawa-sh) [inline config/initializers/sorcery.rb]: **[rubocop]**
- Style/EmptyLines: Extra blank line detected.

File: config/initializers/sorcery.rb [hunk 6]
```diff
@@ -0,0 +1,464 @@
...
+  # Allow the remember_me cookie to be set through AJAX
+  # Default: `true`
+  #
+  # config.remember_me_httponly =
+
+
```
Comment (segawa-sh) [inline config/initializers/sorcery.rb]: **[rubocop]**
- Style/EmptyLines: Extra blank line detected.

File: config/initializers/sorcery.rb [hunk 7]
```diff
@@ -0,0 +1,464 @@
...
+  # -- session timeout --
+  # How long in seconds to keep the session alive.
+  # Default: `3600`
+  #
+  # config.session_timeout =
+
+
```
Comment (segawa-sh) [inline config/initializers/sorcery.rb]: **[rubocop]**
- Style/EmptyLines: Extra blank line detected.

File: config/initializers/sorcery.rb [hunk 8]
```diff
@@ -0,0 +1,464 @@
...
+  # Use the last action as the beginning of session timeout.
+  # Default: `false`
+  #
+  # config.session_timeout_from_last_action =
+
+
```
Comment (segawa-sh) [inline config/initializers/sorcery.rb]: **[rubocop]**
- Style/EmptyLines: Extra blank line detected.

File: config/initializers/sorcery.rb [hunk 9]
```diff
@@ -0,0 +1,464 @@
...
+  # -- http_basic_auth --
+  # What realm to display for which controller name. For example {"My App" => "Application"}
+  # Default: `{"application" => "Application"}`
+  #
+  # config.controller_to_realm_map =
+
+
```
Comment (segawa-sh) [inline config/initializers/sorcery.rb]: **[rubocop]**
- Style/EmptyLines: Extra blank line detected.

File: config/initializers/sorcery.rb [hunk 10]
```diff
@@ -0,0 +1,464 @@
...
+  # -- activity logging --
+  # will register the time of last user login, every login.
+  # Default: `true`
+  #
+  # config.register_login_time =
+
+
```
Comment (segawa-sh) [inline config/initializers/sorcery.rb]: **[rubocop]**
- Style/EmptyLines: Extra blank line detected.

File: config/initializers/sorcery.rb [hunk 11]
```diff
@@ -0,0 +1,464 @@
...
+  # will register the time of last user logout, every logout.
+  # Default: `true`
+  #
+  # config.register_logout_time =
+
+
```
Comment (segawa-sh) [inline config/initializers/sorcery.rb]: **[rubocop]**
- Style/EmptyLines: Extra blank line detected.

File: config/initializers/sorcery.rb [hunk 12]
```diff
@@ -0,0 +1,464 @@
...
+  # will register the time of last user action, every action.
+  # Default: `true`
+  #
+  # config.register_last_activity_time =
+
+
```
Comment (segawa-sh) [inline config/initializers/sorcery.rb]: **[rubocop]**
- Style/EmptyLines: Extra blank line detected.

File: config/initializers/sorcery.rb [hunk 13]
```diff
@@ -0,0 +1,464 @@
...
+  # -- external --
+  # What providers are supported by this app, i.e. [:twitter, :facebook, :github, :linkedin, :xing, :google, :liveid, :salesforce] .
+  # Default: `[]`
+  #
+  # config.external_providers =
+
+
```
Comment (segawa-sh) [inline config/initializers/sorcery.rb]: **[rubocop]**
- Style/EmptyLines: Extra blank line detected.

File: config/initializers/sorcery.rb [hunk 14]
```diff
@@ -0,0 +1,464 @@
...
+  # You can change it by your local ca_file. i.e. '/etc/pki/tls/certs/ca-bundle.crt'
+  # Path to ca_file. By default use a internal ca-bundle.crt.
+  # Default: `'path/to/ca_file'`
+  #
+  # config.ca_file =
+
+
```
Comment (segawa-sh) [inline config/initializers/sorcery.rb]: **[rubocop]**
- Style/EmptyLines: Extra blank line detected.

File: config/initializers/sorcery.rb [hunk 15]
```diff
@@ -0,0 +1,464 @@
...
+  # --- user config ---
+  config.user_config do |user|
+    # -- core --
+    # specify username attributes, for example: [:username, :email].
+    # Default: `[:email]`
+    #
+    # user.username_attribute_names =
+
+
```
Comment (segawa-sh) [inline config/initializers/sorcery.rb]: **[rubocop]**
- Style/EmptyLines: Extra blank line detected.

File: config/initializers/sorcery.rb [hunk 16]
```diff
@@ -0,0 +1,464 @@
...
+    # change *virtual* password attribute, the one which is used until an encrypted one is generated.
+    # Default: `:password`
+    #
+    # user.password_attribute_name =
+
+
```
Comment (segawa-sh) [inline config/initializers/sorcery.rb]: **[rubocop]**
- Style/EmptyLines: Extra blank line detected.

File: config/initializers/sorcery.rb [hunk 17]
```diff
@@ -0,0 +1,464 @@
...
+    # downcase the username before trying to authenticate, default is false
+    # Default: `false`
+    #
+    # user.downcase_username_before_authenticating =
+
+
```
Comment (segawa-sh) [inline config/initializers/sorcery.rb]: **[rubocop]**
- Style/EmptyLines: Extra blank line detected.

File: config/initializers/sorcery.rb [hunk 18]
```diff
@@ -0,0 +1,464 @@
...
+    # change default email attribute.
+    # Default: `:email`
+    #
+    # user.email_attribute_name =
+
+
```
Comment (segawa-sh) [inline config/initializers/sorcery.rb]: **[rubocop]**
- Style/EmptyLines: Extra blank line detected.

File: config/initializers/sorcery.rb [hunk 19]
```diff
@@ -0,0 +1,464 @@
...
+    # change default crypted_password attribute.
+    # Default: `:crypted_password`
+    #
+    # user.crypted_password_attribute_name =
+
+
```
Comment (segawa-sh) [inline config/initializers/sorcery.rb]: **[rubocop]**
- Style/EmptyLines: Extra blank line detected.

File: config/initializers/sorcery.rb [hunk 20]
```diff
@@ -0,0 +1,464 @@
...
+    # what pattern to use to join the password with the salt
+    # Default: `""`
+    #
+    # user.salt_join_token =
+
+
```
Comment (segawa-sh) [inline config/initializers/sorcery.rb]: **[rubocop]**
- Style/EmptyLines: Extra blank line detected.

File: config/initializers/sorcery.rb [hunk 21]
```diff
@@ -0,0 +1,464 @@
...
+    # change default salt attribute.
+    # Default: `:salt`
+    #
+    # user.salt_attribute_name =
+    user.salt_attribute_name = nil
+
+
```
Comment (segawa-sh) [inline config/initializers/sorcery.rb]: **[rubocop]**
- Style/EmptyLines: Extra blank line detected.

File: config/initializers/sorcery.rb [hunk 22]
```diff
@@ -0,0 +1,464 @@
...
+    # how many times to apply encryption to the password.
+    # Default: `nil`
+    #
+    # user.stretches =
+
+
```
Comment (segawa-sh) [inline config/initializers/sorcery.rb]: **[rubocop]**
- Style/EmptyLines: Extra blank line detected.

File: config/initializers/sorcery.rb [hunk 23]
```diff
@@ -0,0 +1,464 @@
...
+    # encryption key used to encrypt reversible encryptions such as AES256.
+    # WARNING: If used for users' passwords, changing this key will leave passwords undecryptable!
+    # Default: `nil`
+    #
+    # user.encryption_key =
+
+
```
Comment (segawa-sh) [inline config/initializers/sorcery.rb]: **[rubocop]**
- Style/EmptyLines: Extra blank line detected.

File: config/initializers/sorcery.rb [hunk 24]
```diff
@@ -0,0 +1,464 @@
...
+    # use an external encryption class.
+    # Default: `nil`
+    #
+    # user.custom_encryption_provider =
+
+
```
Comment (segawa-sh) [inline config/initializers/sorcery.rb]: **[rubocop]**
- Style/EmptyLines: Extra blank line detected.

File: config/initializers/sorcery.rb [hunk 25]
```diff
@@ -0,0 +1,464 @@
...
+    # encryption algorithm name. See 'encryption_algorithm=' for available options.
+    # Default: `:bcrypt`
+    #
+    # user.encryption_algorithm =
+
+
```
Comment (segawa-sh) [inline config/initializers/sorcery.rb]: **[rubocop]**
- Style/EmptyLines: Extra blank line detected.

File: config/initializers/sorcery.rb [hunk 26]
```diff
@@ -0,0 +1,464 @@
...
+    # make this configuration inheritable for subclasses. Useful for ActiveRecord's STI.
+    # Default: `false`
+    #
+    # user.subclasses_inherit_config =
+
+
```
Comment (segawa-sh) [inline config/initializers/sorcery.rb]: **[rubocop]**
- Style/EmptyLines: Extra blank line detected.

File: config/initializers/sorcery.rb [hunk 27]
```diff
@@ -0,0 +1,464 @@
...
+    # -- remember_me --
+    # How long in seconds the session length will be
+    # Default: `604800`
+    #
+    # user.remember_me_for =
+
+
```
Comment (segawa-sh) [inline config/initializers/sorcery.rb]: **[rubocop]**
- Style/EmptyLines: Extra blank line detected.

File: config/initializers/sorcery.rb [hunk 28]
```diff
@@ -0,0 +1,464 @@
...
+    # -- user_activation --
+    # the attribute name to hold activation state (active/pending).
+    # Default: `:activation_state`
+    #
+    # user.activation_state_attribute_name =
+
+
```
Comment (segawa-sh) [inline config/initializers/sorcery.rb]: **[rubocop]**
- Style/EmptyLines: Extra blank line detected.

File: config/initializers/sorcery.rb [hunk 29]
```diff
@@ -0,0 +1,464 @@
...
+    # the attribute name to hold activation code (sent by email).
+    # Default: `:activation_token`
+    #
+    # user.activation_token_attribute_name =
+
+
```
Comment (segawa-sh) [inline config/initializers/sorcery.rb]: **[rubocop]**
- Style/EmptyLines: Extra blank line detected.

File: config/initializers/sorcery.rb [hunk 30]
```diff
@@ -0,0 +1,464 @@
...
+    # the attribute name to hold activation code expiration date.
+    # Default: `:activation_token_expires_at`
+    #
+    # user.activation_token_expires_at_attribute_name =
+
+
```
Comment (segawa-sh) [inline config/initializers/sorcery.rb]: **[rubocop]**
- Style/EmptyLines: Extra blank line detected.

File: config/initializers/sorcery.rb [hunk 31]
```diff
@@ -0,0 +1,464 @@
...
+    # how many seconds before the activation code expires. nil for never expires.
+    # Default: `nil`
+    #
+    # user.activation_token_expiration_period =
+
+
```
Comment (segawa-sh) [inline config/initializers/sorcery.rb]: **[rubocop]**
- Style/EmptyLines: Extra blank line detected.

File: config/initializers/sorcery.rb [hunk 32]
```diff
@@ -0,0 +1,464 @@
...
+    # your mailer class. Required.
+    # Default: `nil`
+    #
+    # user.user_activation_mailer =
+
+
```
Comment (segawa-sh) [inline config/initializers/sorcery.rb]: **[rubocop]**
- Style/EmptyLines: Extra blank line detected.

File: config/initializers/sorcery.rb [hunk 33]
```diff
@@ -0,0 +1,464 @@
...
+    # when true sorcery will not automatically
+    # email activation details and allow you to
+    # manually handle how and when email is sent.
+    # Default: `false`
+    #
+    # user.activation_mailer_disabled =
+
+
```
Comment (segawa-sh) [inline config/initializers/sorcery.rb]: **[rubocop]**
- Style/EmptyLines: Extra blank line detected.

File: config/initializers/sorcery.rb [hunk 34]
```diff
@@ -0,0 +1,464 @@
...
+    # activation needed email method on your mailer class.
+    # Default: `:activation_needed_email`
+    #
+    # user.activation_needed_email_method_name =
+
+
```
Comment (segawa-sh) [inline config/initializers/sorcery.rb]: **[rubocop]**
- Style/EmptyLines: Extra blank line detected.

File: config/initializers/sorcery.rb [hunk 35]
```diff
@@ -0,0 +1,464 @@
...
+    # activation success email method on your mailer class.
+    # Default: `:activation_success_email`
+    #
+    # user.activation_success_email_method_name =
+
+
```
Comment (segawa-sh) [inline config/initializers/sorcery.rb]: **[rubocop]**
- Style/EmptyLines: Extra blank line detected.

File: config/initializers/sorcery.rb [hunk 36]
```diff
@@ -0,0 +1,464 @@
...
+    # do you want to prevent or allow users that did not activate by email to login?
+    # Default: `true`
+    #
+    # user.prevent_non_active_users_to_login =
+
+
```
Comment (segawa-sh) [inline config/initializers/sorcery.rb]: **[rubocop]**
- Style/EmptyLines: Extra blank line detected.

File: config/initializers/sorcery.rb [hunk 37]
```diff
@@ -0,0 +1,464 @@
...
+    # -- reset_password --
+    # reset password code attribute name.
+    # Default: `:reset_password_token`
+    #
+    # user.reset_password_token_attribute_name =
+
+
```
Comment (segawa-sh) [inline config/initializers/sorcery.rb]: **[rubocop]**
- Style/EmptyLines: Extra blank line detected.

File: config/initializers/sorcery.rb [hunk 38]
```diff
@@ -0,0 +1,464 @@
...
+    # expires at attribute name.
+    # Default: `:reset_password_token_expires_at`
+    #
+    # user.reset_password_token_expires_at_attribute_name =
+
+
```
Comment (segawa-sh) [inline config/initializers/sorcery.rb]: **[rubocop]**
- Style/EmptyLines: Extra blank line detected.

File: config/initializers/sorcery.rb [hunk 39]
```diff
@@ -0,0 +1,464 @@
...
+    # when was email sent, used for hammering protection.
+    # Default: `:reset_password_email_sent_at`
+    #
+    # user.reset_password_email_sent_at_attribute_name =
+
+
```
Comment (segawa-sh) [inline config/initializers/sorcery.rb]: **[rubocop]**
- Style/EmptyLines: Extra blank line detected.

File: config/initializers/sorcery.rb [hunk 40]
```diff
@@ -0,0 +1,464 @@
...
+    # mailer class. Needed.
+    # Default: `nil`
+    #
+    # user.reset_password_mailer =
+
+
```
Comment (segawa-sh) [inline config/initializers/sorcery.rb]: **[rubocop]**
- Style/EmptyLines: Extra blank line detected.

File: config/initializers/sorcery.rb [hunk 41]
```diff
@@ -0,0 +1,464 @@
...
+    # reset password email method on your mailer class.
+    # Default: `:reset_password_email`
+    #
+    # user.reset_password_email_method_name =
+
+
```
Comment (segawa-sh) [inline config/initializers/sorcery.rb]: **[rubocop]**
- Style/EmptyLines: Extra blank line detected.

File: config/initializers/sorcery.rb [hunk 42]
```diff
@@ -0,0 +1,464 @@
...
+    # when true sorcery will not automatically
+    # email password reset details and allow you to
+    # manually handle how and when email is sent
+    # Default: `false`
+    #
+    # user.reset_password_mailer_disabled =
+
+
```
Comment (segawa-sh) [inline config/initializers/sorcery.rb]: **[rubocop]**
- Style/EmptyLines: Extra blank line detected.

File: config/initializers/sorcery.rb [hunk 43]
```diff
@@ -0,0 +1,464 @@
...
+    # how many seconds before the reset request expires. nil for never expires.
+    # Default: `nil`
+    #
+    # user.reset_password_expiration_period =
+
+
```
Comment (segawa-sh) [inline config/initializers/sorcery.rb]: **[rubocop]**
- Style/EmptyLines: Extra blank line detected.

File: config/initializers/sorcery.rb [hunk 44]
```diff
@@ -0,0 +1,464 @@
...
+    # hammering protection, how long in seconds to wait before allowing another email to be sent.
+    # Default: `5 * 60`
+    #
+    # user.reset_password_time_between_emails =
+
+
```
Comment (segawa-sh) [inline config/initializers/sorcery.rb]: **[rubocop]**
- Style/EmptyLines: Extra blank line detected.

File: config/initializers/sorcery.rb [hunk 45]
```diff
@@ -0,0 +1,464 @@
...
+    # -- brute_force_protection --
+    # Failed logins attribute name.
+    # Default: `:failed_logins_count`
+    #
+    # user.failed_logins_count_attribute_name =
+
+
```
Comment (segawa-sh) [inline config/initializers/sorcery.rb]: **[rubocop]**
- Style/EmptyLines: Extra blank line detected.

File: config/initializers/sorcery.rb [hunk 46]
```diff
@@ -0,0 +1,464 @@
...
+    # This field indicates whether user is banned and when it will be active again.
+    # Default: `:lock_expires_at`
+    #
+    # user.lock_expires_at_attribute_name =
+
+
```
Comment (segawa-sh) [inline config/initializers/sorcery.rb]: **[rubocop]**
- Style/EmptyLines: Extra blank line detected.

File: config/initializers/sorcery.rb [hunk 47]
```diff
@@ -0,0 +1,464 @@
...
+    # How many failed logins allowed.
+    # Default: `50`
+    #
+    # user.consecutive_login_retries_amount_limit =
+
+
```
Comment (segawa-sh) [inline config/initializers/sorcery.rb]: **[rubocop]**
- Style/EmptyLines: Extra blank line detected.

File: config/initializers/sorcery.rb [hunk 48]
```diff
@@ -0,0 +1,464 @@
...
+    #
+    # user.unlock_token_mailer = UserMailer
+
+    # -- activity logging --
+    # Last login attribute name.
+    # Default: `:last_login_at`
+    #
+    # user.last_login_at_attribute_name =
+
+
```
Comment (segawa-sh) [inline config/initializers/sorcery.rb]: **[rubocop]**
- Style/EmptyLines: Extra blank line detected.

File: config/initializers/sorcery.rb [hunk 49]
```diff
@@ -0,0 +1,464 @@
...
+    # Last logout attribute name.
+    # Default: `:last_logout_at`
+    #
+    # user.last_logout_at_attribute_name =
+
+
```
Comment (segawa-sh) [inline config/initializers/sorcery.rb]: **[rubocop]**
- Style/EmptyLines: Extra blank line detected.

File: config/initializers/sorcery.rb [hunk 50]
```diff
@@ -0,0 +1,464 @@
...
+    # Last activity attribute name.
+    # Default: `:last_activity_at`
+    #
+    # user.last_activity_at_attribute_name =
+
+
```
Comment (segawa-sh) [inline config/initializers/sorcery.rb]: **[rubocop]**
- Style/EmptyLines: Extra blank line detected.

File: config/initializers/sorcery.rb [hunk 51]
```diff
@@ -0,0 +1,464 @@
...
+    # How long since last activity is the user defined logged out?
+    # Default: `10 * 60`
+    #
+    # user.activity_timeout =
+
+
```
Comment (segawa-sh) [inline config/initializers/sorcery.rb]: **[rubocop]**
- Style/EmptyLines: Extra blank line detected.

File: config/initializers/sorcery.rb [hunk 52]
```diff
@@ -0,0 +1,464 @@
...
+    # -- external --
+    # Class which holds the various external provider data for this user.
+    # Default: `nil`
+    #
+    # user.authentications_class =
+
+
```
Comment (segawa-sh) [inline config/initializers/sorcery.rb]: **[rubocop]**
- Style/EmptyLines: Extra blank line detected.

File: config/initializers/sorcery.rb [hunk 53]
```diff
@@ -0,0 +1,464 @@
...
+    # User's identifier in authentications class.
+    # Default: `:user_id`
+    #
+    # user.authentications_user_id_attribute_name =
+
+
```
Comment (segawa-sh) [inline config/initializers/sorcery.rb]: **[rubocop]**
- Style/EmptyLines: Extra blank line detected.

File: config/initializers/sorcery.rb [hunk 54]
```diff
@@ -0,0 +1,464 @@
...
+    # Provider's identifier in authentications class.
+    # Default: `:provider`
+    #
+    # user.provider_attribute_name =
+
+
```
Comment (segawa-sh) [inline config/initializers/sorcery.rb]: **[rubocop]**
- Style/EmptyLines: Extra blank line detected.

File: config/initializers/sorcery.rb [hunk 55]
```diff
@@ -0,0 +1,464 @@
...
+    # User's external unique identifier in authentications class.
+    # Default: `:uid`
+    #
+    # user.provider_uid_attribute_name =
+  end
+
+  # This line must come after the 'user config' block.
+  # Define which model authenticates with sorcery.
+  config.user_class = "User"
```
Comment (segawa-sh) [inline config/initializers/sorcery.rb]: **[rubocop]**
- Style/StringLiterals: Prefer single-quoted strings when you don't need string interpolation or special symbols. [:ref](https://github.com/bbatsov/ruby-style-guide#consistent-string-literals)

**Comment by segawa-sh** [comment]
Click here if you want to create rubocop's auto-correct Pull Request 
[](https://www.sideci.com/github_repositories/47532393/auto_correct/rubocop/pull_request/9/branch/tektoh%2FISSUE-6)

See the documentation if you want to get more information about auto-correct.
[Document](https://www.sideci.com/ja/docs/autofix-rubocop)

File: app/views/kaminari/_paginator.html.erb (hunk 1 above)
Comment (segawa-sh) [inline app/views/kaminari/_paginator.html.erb:5]: **[rails_best_practices]**
- move code into model (page use_count > 2) [:ref](http://rails-bestpractices.com/posts/2010/07/24/move-code-into-model/)

**Pull request merged**